File: week5_lab/echon.pl
#!/usr/bin/perl -w

if($#ARGV != 1){
        die "Usage: ./echon.pl <number of lines> <string>\n";


}

$num=$ARGV[0];
$string=$ARGV[1];
if($num =~ /^\d+$/){

if($num >= 0){
        for($i=0;$i<$num;$i++){
                print "$string\n";
        }
}
}
else{
        print "./echon.pl: argument 1 must be a non-negative integer\n";
}

File: week6_lab/echon.py
#!/usr/bin/python3

import sys

if len(sys.argv) != 3:
        print("Usage: ./echon.py <number of lines> <string>")
elif (sys.argv[1]).isdigit() and int(sys.argv[1]) >= 0:
        print((sys.argv[2] + "\n") * int(sys.argv[1]), end = "")
else:
        print("./echon.py: argument 1 must be a non-negative integer")
